### PR TITLE
Issue 38 prototype

### PR DESCRIPTION
**Pull request by dploeger**
@ThmsKnz I've opened this PR for better communicating about the changes and to better see the differences.

### CODE REVIEW

**Comment by dploeger** [comment]
Update cache dialog now works great 👍🏽

File: addons/egoventure/cache/scene_cache.gd [hunk 1]
```diff
@@ -29,6 +29,29 @@ var _queued_items: Array = []
 # Do not remove these scenes from the cache
 var _permanent_cache: PoolStringArray = []
 
+# Cache Map
+var _cache_map: CacheMap
+
+# Estimated Cache Size
+var _cache_size: int
+var _cache_max_size = 51200 # for prototype: set cache max size to 50 MB
```
Comment (dploeger) [inline addons/egoventure/cache/scene_cache.gd]: TODO for later: Make this configurable

File: addons/egoventure/cache/scene_cache.gd [hunk 2]
```diff
@@ -95,74 +127,105 @@ func get_scene(path: String) -> PackedScene:
...
+				var mapped_scene_size = _cache_map.map[mapped_scene][0]
+				while (
+					_cache_size + mapped_scene_size > _cache_max_size and
+					_cache_age - 1 != _cache_clear_age  # don't clear cache that was loaded in previous step
+					):
+					print("Current cache size %s + new scene size %s > max cache size %s" % [_cache_size, mapped_scene_size, _cache_max_size])
+					print("Remove scenes from cache with age %s" % _cache_clear_age)
+					_remove_scenes_from_cache(_cache_clear_age)
+					_cache_clear_age += 1
+					#print ("Cache Clear Age has been increased to: %s" % _cache_clear_age)
```
Comment (dploeger) [inline addons/egoventure/cache/scene_cache.gd]: Remove commented-out code

**Comment by ThmsKnz** [comment]
This pull request can be closed/deleted. I've replaced it with a pull request from branch 'issue-38'

**Pull request closed without merging**